apps/aevm/src/aevm_eeevm_state.erl

```diff
@@ -84,6 +84,11 @@ init(#{ env  := Env
 
     ChainState = maps:get(chainState, Env),
     ChainAPI =  maps:get(chainAPI, Env),
+    {IsOffChain, OnChainTrees} =
+        case maps:get(off_chain, Env, false) of
```

Comment:
Avoiding default would make the code more readable, by making explicit the point of the decision about off/on chain.